### PR TITLE
Enhancement: Open local service, locally

### PR DESCRIPTION
**Pull request by 304NotModified**
## What Changed?

- Service local installed, should open without UNC path.
- deduplicated code
- moved isLocalMachine to helper (and renamed currentMachine to localMachine - it think that's better)

Before:

image

image

after:

image

## Issues Closed
none

### CODE REVIEW

**Comment by 304NotModified** [comment]
done with pushes ;)

Please review, feel free to comment on names

File: src/ServiceBouncer/MainForm.cs [hunk 1]
```diff
@@ -334,9 +334,8 @@ private async Task Connect()
                 toolStripConnectButton.Tag = @"Connected";
                 toolStripConnectButton.Image = Properties.Resources.Disconnect;
 
-                machineHostname = machineHostname.Trim();
-                var isCurrentMachine = machineHostname == "." || Environment.MachineName.Equals(machineHostname, StringComparison.CurrentCultureIgnoreCase);
-                backgroundRefreshSeconds = isCurrentMachine ? 1 : 30;
+                var isLocalMachine = EnvHelper.IsLocalMachine(machineHostname);
```
Comment (BlythMeister) [inline src/ServiceBouncer/MainForm.cs]: Could we remove the isLocalMachine variable here?

File: src/ServiceBouncer/EnvHelper.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        /// <summary>
+        /// Is this the current local machine?
+        /// </summary>
+        /// <param name="machineHostname"></param>
+        /// <returns></returns>
+        public static bool IsLocalMachine(string machineHostname)
+        {
+            machineHostname = machineHostname.Trim();
+            var isLocalMachine = machineHostname == "." || Environment.MachineName.Equals(machineHostname, StringComparison.CurrentCultureIgnoreCase);
```
Comment (BlythMeister) [inline src/ServiceBouncer/EnvHelper.cs]: Could we remove the isLocalMachine variable here?

File: src/ServiceBouncer/Extensions/ServiceControllerExtensions.cs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 using System.IO;
 using System.Management;
+using System.Runtime.InteropServices;
```
Comment (BlythMeister) [inline src/ServiceBouncer/Extensions/ServiceControllerExtensions.cs]: What is InteropServices added for?

**Review by BlythMeister — CHANGES_REQUESTED** [review]

File: src/ServiceBouncer/MainForm.cs (hunk 1 above)
Comment (304NotModified) [reply src/ServiceBouncer/MainForm.cs]: done

File: src/ServiceBouncer/EnvHelper.cs (hunk 2 above)
Comment (304NotModified) [reply src/ServiceBouncer/EnvHelper.cs]: done

File: src/ServiceBouncer/Extensions/ServiceControllerExtensions.cs (hunk 3 above)
Comment (304NotModified) [reply src/ServiceBouncer/Extensions/ServiceControllerExtensions.cs]: good catch, fixed

**Comment by 304NotModified** [comment]
changes pushed! 

(if it needs a squash or something like that, just let me know)

**Comment by 304NotModified** [comment]
changes applied and pushed!

(if it needs a squash or something like that, just let me know)

**Review by BlythMeister — APPROVED** [review]
Perfect Thanks

**Pull request merged**